src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -96,7 +96,7 @@ public Message DeserializeMessage(string rawMessage)
             // PERF: This is slow, we deserialize the message, and the payload into JToken just to get the header. We then
             // deserialize the data from the JToken, but that is twice as expensive as deserializing the whole object directly into the final object type.
             // We need this for backward compatibility though.
-            return Deserialize<VersionedMessage>(rawMessage);
+            return Deserialize<VersionedMessage>(rawMessage)!;
```

Comment:
Are we sure to always have a value here? I see that below you are doing a null check on the deserialized data.

Reply:
When unsure, I usually prefer a temp var and a `Debug.Assert` as it would fail and give me some context info instead of potentially failing somewhere else and not realizing because of all our try/catch.

Maybe we could just add a todo comment above so we can try to check null cases later?

Reply:
It worked until now, so imho not really a big problem.